### PR TITLE
VUE-390 Auto Deposit Sign Up Form

### PR DESCRIPTION
**Pull request by eddieferrer**
* Implemented auto deposit sign up form.
* Fix other forms with day of month to avoid GQL validation errors in federation

### CODE REVIEW

**Comment by eddieferrer** [comment]
This is what it looks like so far (this PR doesnt include the drop in, pictured here)
screencapture-dev-vm-01-kiva-org-auto-deposit-new-2020-09-03-15_39_21

File: src/components/AutoDeposit/AutoDepositSignUpForm.vue [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+			donationOptionSelected: '15',
+			isDonationOptionsDirty: false,
+			showDropInPayments: true,
+			isLoggedIn: false,
+		};
+	},
+	mixins: [
+		validationMixin,
+	],
+	validations: {
```
Comment (eddieferrer) [inline src/components/AutoDeposit/AutoDepositSignUpForm.vue:208]: I double checked these validation with Leo

File: src/components/AutoDeposit/AutoDepositSignUpForm.vue [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+
+// import AutoDepositDropInPaymentWrapper from '@/components/AutoDeposit/AutoDepositDropInPaymentWrapper';
+import IconPencil from '@/assets/icons/inline/pencil.svg';
+import KvCheckbox from '@/components/Kv/KvCheckbox';
+import KvCurrencyInput from '@/components/Kv/KvCurrencyInput';
+import KvDropdownRounded from '@/components/Kv/KvDropdownRounded';
+
+const pageQuery = gql`query monthlyGoodSetupPageControl {
+    general {
+		braintreeDropInFeature: uiConfigSetting(key: "feature.braintree_dropin") {
```
Comment (mcstover) [inline src/components/AutoDeposit/AutoDepositSignUpForm.vue:170]: I don't think we need this seeing as there wouldn't be any type of fallback...also I wouldn't want it to collide with other drop-in usage elsewhere. We would need to do a series of convoluted steps to be able to redirect back to the old page which I don't think are worth it. Let's just roll with the new new.

File: src/components/AutoDeposit/AutoDepositSignUpForm.vue [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+				max="31"
+				v-model.number="dayOfMonth"
+			>
+			<button
+				class="ad-sign-up-form__button--ordinal-day"
+				@click="isDayInputShown = true"
+				v-if="!isDayInputShown"
+			>
+				<strong>{{ dayOfMonth | numeral('Oo') }}</strong>
+				<icon-pencil class="icon-pencil" />
```
Comment (ryan-ludwig) [inline src/components/AutoDeposit/AutoDepositSignUpForm.vue:43]: The end result will be pretty much the same but I think you can use` <kv-icon name="pencil">` for this

**Review by mcstover — APPROVED** [review]

File: src/components/AutoDeposit/AutoDepositSignUpForm.vue (hunk 3 above)
Comment (eddieferrer) [reply src/components/AutoDeposit/AutoDepositSignUpForm.vue:43]: Good catch, I used this in a couple of places. I'll fix in a follow up PR.

**Pull request merged**